[AST] update card CD management plus wording

otocephaly opened the pull request:
## Pull request type

- [x] This is a bugfix to existing functionality

## Pull request details
- [x] This is in response to a discussion or thread on Discord: *[Link to thread or start of discussion]*(https://www.fflogs.com/reports/a:LhGYBmQVjwpJWKqb?fight=20&source=75)
- [x] The goal of this PR is detailed below:
    - Update amount of cards to align with divination burst windows rather than keeping always on CD

## Testing / Validation
- [x] I used the log(s) listed below to develop and test this bugfix:
    - https://www.fflogs.com/reports/a:LhGYBmQVjwpJWKqb?fight=20&source=75
    - (overwritting example) https://www.fflogs.com/reports/a:N418xZWmrDX2aKBR?fight=25&source=359

   *[Provide a list of log links - either direct to FFLogs or the xivanalysis log link for development or production]*

## Job Maintenance
- [x] I have collaborated with one or more job maintainers for the job(s) in this PR while developing it

## Screenshots
Updated wording and logic
![image](https://github.com/user-attachments/assets/44c51385-7aa3-45a7-8436-bd6e5bce1338)

Overwrite example and updated macrocosmos wording
![image](https://github.com/user-attachments/assets/20f99a70-f756-486a-a8a7-4e7e06cbdbed)

Old for reference
![image](https://github.com/user-attachments/assets/147f58e5-9be0-425d-a684-59baee7762fe)
![image](https://github.com/user-attachments/assets/b560a3fd-4c6d-4d8c-960e-615c06610da4)

otocephaly committed:
[AST] update card CD management plus wording

AkairyuGestalter commented:
I still feel like there's a better way to represent the intent we're trying to get at here. The cooldown fudging to only ask for a certain number of Draw uses based on alignment with Divination's CD makes sense, but the checklist requirements for using Play 1 and Lord are already being handled by the ExpectedCardEvaluator as part of the Divination RaidBuffWindow analysis. I feel like what we're actually missing to account for the "you might intentionally drift outside of the window because of fight timeline reasons" case, is a warning against overwriting Lord/Play 1 if not used when the next Draw is pressed, rather than trying to stuff to much into this checklist's requirements.

If you'd rather we keep this as-is at least for now, let me know and I'll go ahead and merge it in since it is doing what it was designed to do. I just wanted to bring up the alternative while I was thinking about it :)

otocephaly commented:
> I still feel like there's a better way to represent the intent we're trying to get at here. The cooldown fudging to only ask for a certain number of Draw uses based on alignment with Divination's CD makes sense, but the checklist requirements for using Play 1 and Lord are already being handled by the ExpectedCardEvaluator as part of the Divination RaidBuffWindow analysis. I feel like what we're actually missing to account for the "you might intentionally drift outside of the window because of fight timeline reasons" case, is a warning against overwriting Lord/Play 1 if not used when the next Draw is pressed, rather than trying to stuff to much into this checklist's requirements.
> 
> If you'd rather we keep this as-is at least for now, let me know and I'll go ahead and merge it in since it is doing what it was designed to do. I just wanted to bring up the alternative while I was thinking about it :)

hmm I see what you're getting at. the only edge case it covers that raidbuffwindow does not is when a boss is killed between the odd minute and even minute where in that case, you'd be better off just using the melee card and lord during the 1 min burst. The overwriting is a good suggestion, let me take that back and put something in.
Do you think it'd be better to remove the checklist portion and only put a recommendation for when the boss is killed within 1 min of the expected divination cast and the balance/lord is held? (an example of this happening is M5S since the boss even during enrage is always killed just before raidbuffs can go out, could put as a medium or minor suggestion since sometimes you can't predict if the boss will be killed before raid buffs)

AkairyuGestalter commented:
> hmm I see what you're getting at. the only edge case it covers that raidbuffwindow does not is when a boss is killed between the odd minute and even minute where in that case, you'd be better off just using the melee card and lord during the 1 min burst. The overwriting is a good suggestion, let me take that back and put something in. Do you think it'd be better to remove the checklist portion and only put a recommendation for when the boss is killed within 1 min of the expected divination cast and the balance/lord is held? (an example of this happening is M5S since the boss even during enrage is always killed just before raidbuffs can go out, could put as a medium or minor suggestion since sometimes you can't predict if the boss will be killed before raid buffs)

I think keeping the checklist with just the "did you use Draw enough" requirement, but leaving off the "use Play 1" and "use Lord" requirements since those are covered by Divination is good. I know we're usually hesitant about being too strict about ending the fight with resources still available since killtimes are variable, so if you choose to add something there, I'd probably make it only ever a Minor suggestion.

otocephaly committed:
updated draw, overwrite, and macrocosmos

otocephaly commented:
Hi @AkairyuGestalter ,
I've updated per your suggestions (the screenshots above have been updated)
Additionally, I updated macrocosmos usages to have higher thresholds and better wording to not imply it should be used on CD (it's mostly just to suggest that people use it more often in the fight)

AkairyuGestalter commented on src/parser/jobs/ast/modules/Draw.tsx:
```diff
@@ -139,33 +159,24 @@ export class Draw extends Analyser {
 			displayOrder: DISPLAY_ORDER.DRAW_CHECKLIST,
 			name: <Trans id="ast.draw.checklist.name">
-				Play as many cards as possible
+				Play as many cards as possible considering <DataLink action="DIVINATION" /> windows.
```
Probably want to revise the checklist name and description now that it's truly focusing on Draw usage. Otherwise lgtm.

otocephaly replied:
good catch! thank you
updated

otocephaly committed:
updated based on feedback

AkairyuGestalter reviewed (approved):
I like it. Away we go.
